libethash-cl/ethash_cl_miner.cpp

```diff
@@ -346,26 +326,23 @@ bool ethash_cl_miner::init(
 			ETHCL_LOG(program.getBuildInfo<CL_PROGRAM_BUILD_LOG>(device).c_str());
 			return false;
 		}
-		if (m_dagChunksNum == 1)
-		{
-			ETHCL_LOG("Loading single big chunk kernels");
-			m_hash_kernel = cl::Kernel(program, "ethash_hash");
-			m_search_kernel = cl::Kernel(program, "ethash_search");
-		}
-		else
-		{
-			ETHCL_LOG("Loading chunk kernels");
-			m_hash_kernel = cl::Kernel(program, "ethash_hash_chunks");
-			m_search_kernel = cl::Kernel(program, "ethash_search_chunks");
-		}
 
 		// create buffer for dag
-		if (m_dagChunksNum == 1)
+		try
 		{
-			ETHCL_LOG("Creating one big buffer");
+			m_dagChunksNum = 1;
 			m_dagChunks.push_back(cl::Buffer(m_context, CL_MEM_READ_ONLY, _dagSize));
+			ETHCL_LOG("Created one big buffer for the DAG");
 		}
-		else
+		catch (...)
+		{
+			cl_ulong result;
+			device.getInfo(CL_DEVICE_MAX_MEM_ALLOC_SIZE, &result);
+			ETHCL_LOG(
+				"Failed to allocate 1 big chunk. Max allocateable memory is "
+				<< result << ". Trying to allocate 4 chunks."
+			);
+			m_dagChunksNum = 4;
```

Comment:
Because it's a "Safe" value. We probably need to at some point soon make it dynamic but the problem with this is that we would need to be rewritting the kernel dynamically at every work loop too.